include/deal.II/matrix_free/cuda_matrix_free.h

```diff
@@ -206,6 +248,91 @@ namespace CUDAWrappers
     memory_consumption() const;
 
   private:
+    /**
+     * Helper function. Loop over all the cells and apply the functor on each
+     * element in parallel. This function is used when MPI is not used.
+     */
+    template <typename functor, typename VectorType>
+    void
+    serial_cell_loop(const functor &   func,
+                     const VectorType &src,
+                     VectorType &      dst) const;
+
+    /**
+     * Helper function. Loop over all the cells and apply the functor on each
+     * element in parallel. This function is used when MPI is used.
+     */
+    template <typename functor>
+    void
+    distributed_cell_loop(
+      const functor &                                                      func,
+      const LinearAlgebra::distributed::Vector<Number, MemorySpace::CUDA> &src,
+      LinearAlgebra::distributed::Vector<Number, MemorySpace::CUDA> &dst) const;
+
+    /**
+     * This function should never be called. Calling it results in an internal
```

Comment:
maybe mention why we need it/ why it's here?

Reply:
It's needed to compile because `cell_loop` can be instantiated for both `LA::d::Vector` and `CUDA::Vector` but the implementation of `distributed_cell_loop` uses functions that `CUDA::Vector` does not have.

Reply:
i guessed so, i just suggest to somehow mention it (in a short form) in the documentation.

Reply:
I did add a sentence :-)

Reply:
It is `private` anyway.